enzy_htp/structure/structure.py

```diff
@@ -614,6 +614,48 @@ def assign_ncaa_chargespin(self, net_charge_mapper: Dict[str, Tuple[int, int]]):
 
     #endregion
 
+    def get_residue(self, key:str) -> Residue:
+        """Returns the residue specified by a key with the format <chain_name>.<residue_index>. Function does not check
+        for identify of the Residue. Errors and exits if the format is incorrect or the specified Residue does not exist.
+
+        Args:
+            key: specification of the residue to select as a str().
+
+        Returns:
+            The specified Residue.
+        """
+        if key.count('.') != 1:
+            _LOGGER.error(f"Invalid key format in {key}. Expect <chain_name>.<residue_index>. Exiting...")
+            exit( 1 )
+        
+        chain, rnum = key.split('.')
+        rnum = int(rnum) 
+
+        for res in self.residues:
+            if res.parent.name == chain and res.idx == rnum:
+                return res
+        else:
+            _LOGGER.error(f"Unable to locate residue {key} in Structure. Exiting...")
+            exit( 1 )
+
+
+
+    def get_atom(self, key:str) -> Atom:
```

Comment:
I see what you want here.

I would like to design this as `Structure().get(key: str) -> Union[Chain, Residue, Atom]`

Reply:
We can either leave a TODO here or I can help apply these changes if it is too much to change

Reply:
I changed it in a commit